src/optimizer/rule/regex_optimizations.cpp

```diff
@@ -12,17 +14,76 @@ namespace duckdb {
 RegexOptimizationRule::RegexOptimizationRule(ExpressionRewriter &rewriter) : Rule(rewriter) {
 	auto func = make_uniq<FunctionExpressionMatcher>();
 	func->function = make_uniq<SpecificFunctionMatcher>("regexp_matches");
-	func->policy = SetMatcher::Policy::ORDERED;
+	func->policy = SetMatcher::Policy::SOME;
 	func->matchers.push_back(make_uniq<ExpressionMatcher>());
 	func->matchers.push_back(make_uniq<ConstantExpressionMatcher>());
+
 	root = std::move(func);
 }
 
+struct LikeString {
```

Comment:
very much a nitpick, but I've recently started using unnamed namespaces to essentially make struct definitions static that are only relevant to this file.

```c++
namespace {
struct LikeString {
    bool exists = true;
    string lile_string = "";
};
} //namespace
```